kive/container/management/commands/runcontainer.py

```diff
@@ -86,7 +89,12 @@ def fill_sandbox(self, run):
         input_path = os.path.join(run.full_sandbox_path, 'input')
         os.mkdir(input_path)
         for dataset in run.datasets.all():
-            target_path = os.path.join(input_path, dataset.argument.name)
+            if dataset.argument.argtype in (
+                    ContainerArgumentType.OPTIONAL_MULTIPLE_INPUT,
+                    ContainerArgumentType.OPTIONAL_INPUT):
+                target_path = os.path.join(input_path, dataset.dataset.name)
```

Comment:
I'm worried about name collisions here. Would it be safer to add a number to the argument name when needed? To be extra paranoid, keep a set of file names you've created so far, and check all arguments for collisions. That way `--option*` can't collide with `--option2` when you pass in a second file.

Reply:
If the filenames on disk aren't super important, could we just tack a statistically-unique random string to the end of each filename as we copy them in? That lets us handle each file without reference to the others.

Reply:
OK, but we can do better than statistically unique. Each dataset has a unique id, so add that to the file name. Preferably, keep the file extension unchanged by adding the id between the base name and the extension.